Tests/Microsoft.AppCenter.Test.Windows/Storage/StorageAdapterTest.cs

```diff
@@ -26,218 +23,211 @@ public class StorageAdapterTest
             {
-                Assert.Fail("Shouldn't have thrown exception");
+                Assert.IsTrue(e.Message.Contains("Failed to open database connection"));
             }
-            adapter.Dispose();
+        }
+
+        /// <summary>
+        /// Verify that storage adapter is disposed when calling Dispose().
+        /// </summary>
+        [TestMethod]
+        public void StorageAdapterDisposedOnDisposeCall()
+        {
+            InitializeStorageAdapter();
+            _adapter.Dispose();
+            var exceptionThrown = false;
             try
             {
-                // Try get data before database initialize.
-                CreateTableHelper(_adapter);
-                Assert.Fail("Should have thrown exception");
+                CreateTable();
             }
-            catch (Exception e)
+            catch
             {
-                Assert.IsFalse(e.Message.Contains("Should have thrown exception"));
+                exceptionThrown = true;
             }
+            Assert.IsTrue(exceptionThrown);
         }
 
         /// <summary>
-        /// Verify that database is not initilaze.
+        /// Verify that database is not initialized.
         /// </summary>
         [TestMethod]
-        public void DatabaseIsNotInitilazeWhenCallCount()
+        public void DatabaseIsNotInitializedWhenCallCount()
         {
             // Prepare data.
-            var exception = new StorageException("The database wasn't initialized.");
+            var exceptionThrown = false;
             try
             {
-                // Try get data before database initialize.
+                // Try to get data before database initialization.
                 _adapter.Count(TableName, ColumnChannelName, StorageTestChannelName);
-                Assert.Fail("Should have thrown exception");
             }
             catch (Exception e)
             {
-                Assert.AreEqual(exception.Message, e.Message);
+                exceptionThrown = true;
+                Assert.AreEqual("The database wasn't initialized.", e.Message);
             }
+            Assert.IsTrue(exceptionThrown);
+            InitializeStorageAdapter();
+            CreateTable();
             try
             {
-                // Initialize database.
-                _adapter.Initialize(DatabasePath);
-            }
-            catch
-            {
-                // Handle exception, database is not created with Mock.
-            }
-            CreateTableHelper(_adapter);
-            try
-            {
-                // Try get data after database initialize.
+                // Try to get data after database initialization.
                 _adapter.Count(TableName, ColumnChannelName, StorageTestChannelName);
             }
-            catch (Exception e)
+            catch
             {
-                Assert.Fail("Shouldn't have thrown exception");
+                Assert.Fail("Unexpected exception thrown on count.");
```

Comment:
Why not check the returned value of _adapter.Count method? 

When _adapter.Count method throws an exception, why not let it throw and fail the test?

Reply:
Makes sense actually